t/ui/13-admin.t

```diff
@@ -677,17 +677,25 @@ subtest 'Manage API keys' => sub {
     };
 
     subtest 'create key with expiration date' => sub {
-        $driver->find_element('#api-keys-form input[type=submit]')->click;
-        is(scalar @{$driver->find_child_elements($tbody = api_keys_tbody, 'tr')}, 1, 'exactly one key present');
-        like($tbody->get_text, qr/\d{4}-\d{2}-\d{2} \d{2}:\d{2}:\d{2}/, 'new key has expiration date');
+        my @rows = wait_for_element(
+            trigger_function => sub { $driver->find_element('#api-keys-form input[type=submit]')->click },
+            selector => '#api-keys-tbody > tr', description => 'key row present'
+        );
+        is(scalar @rows, 1, 'exactly one key present') or diag explain @rows;
+        like($rows[0]->get_text, qr/\d{4}-\d{2}-\d{2} \d{2}:\d{2}:\d{2}/, 'new key has expiration date');
     };
 
     subtest 'create key without expiration date' => sub {
+        $tbody = api_keys_tbody;
         unlike($tbody->get_text, qr/never/, 'no key without expiration date present so far');
         $driver->find_element_by_id('expiration')->click;
-        $driver->find_element('#api-keys-form input[type=submit]')->click;
-        is(scalar @{$driver->find_child_elements($tbody = api_keys_tbody, 'tr')}, 2, 'two keys present now');
-        like($tbody->get_text, qr/never/, 'new key has expiration date');
+        wait_for_element(
+            trigger_function => sub { $driver->find_element('#api-keys-form input[type=submit]')->click },
+            selector => '#api-keys-tbody > tr', description => 'key row present (without expiration)'
+        );
+        my @rows = $driver->find_elements('#api-keys-tbody > tr');
```

Comment:
With my previous suggestion this repetition could be avoided (and the return value of `wait_for_element` be used instead).